### PR TITLE
sink/mysql: explicit_defaults_for_timestamp compatibility with mysql

### LINKED ISSUE #1585: sink data to mysql5.7, the data is inconsistent
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
# 1.1
tiup cluster deploy tidb-38 v4.0.11 topo.yaml --user ubuntu -i cykey.pem
tiup cluster start tidb-38
tiup cluster patch tidb-38 ./cdc-nightly.tar.gz -N xx.xx.xx.xx:8300
# 1.2
tiup ctl:nightly cdc changefeed create --pd="http://xx.xx.xx.xx:2379" --changefeed-id test-cdc-38 --sink-uri="mysql://root@xx.xx.xx.xx:3306"
# 1.3
./runSQL.sh props.mysql sql.mysql/tableCreates.sql && ./runSQL.sh props.mysql
./runLoader.sh props.mysql
# 1.4
./bin/sync_diff_inspector --config=./config.toml

2. What did you expect to see?
the sync_diff_inspector shows that the data is consistent

3. What did you see instead?
[2021/03/30 10:57:03.280 +00:00] [WARN] [diff.go:893] ["find different row"] [column=ol_delivery_d] [row1="{ ol_supply_w_id: 1, ol_delivery_d: 2021-03-30 07:02:32, ol_amount: 9283.38, ol_o_id: 2182, ol_number: 7, ol_i_id: 7762, ol_quantity: 5, ol_dist_info: AyhG99XdFJ3HbhRBjtFhM5PU, ol_w_id: 1, ol_d_id: 4,  }"] [row2="{ ol_o_id: 2182, ol_i_id: 7762, ol_delivery_d: IsNull, ol_quantity: 5, ol_supply_w_id: 1, ol_dist_info: AyhG99XdFJ3HbhRBjtFhM5PU, ol_w_id: 1, ol_d_id: 4, ol_number: 7, ol_amount: 9283.38,  }"]
[2021/03/30 10:57:03.280 +00:00] [WARN] [diff.go:893] ["find different row"] [column=ol_delivery_d] [row1="{ ol_o_id: 2538, ol_number: 3, ol_i_id: 12445, ol_supply_w_id: 1, ol_quantity: 5, ol_w_id: 1, ol_d_id: 1, ol_delivery_d: 2021-03-30 07:01:53, ol_amount: 512.30, ol_dist_info: lCs1KXiiBwi4SswcrjCkD7tH,  }"] [row2="{ ol_number: 3, ol_delivery_d: IsNull
[cdc.log](https://github.com/pingcap/ticdc/files/6228712/cdc.log)
, ol_amount: 512.30, ol_d_id: 1, ol_o_id: 2538, ol_supply_w_id: 1, ol_quantity: 5, ol_dist_info: lCs1KXiiBwi4SswcrjCkD7tH, ol_w_id: 1, ol_i_id: 12445,  }"]
[2021/03/30 10:57:03.280 +00:00] [INFO] [diff.go:730] ["[update]"] [sql="REPLACE INTO `tpcc`.`bmsql_order_line`(`ol_w_id`,`ol_d_id`,`ol_o_id`,`ol_number`,`ol_i_id`,`ol_delivery_d`,`ol_amount`,`ol_supply_w_id`,`ol_quantity`,`ol_dist_info`) VALUES (1,1,2538,3,12445,'2021-03-30 07:01:53',512.30,1,5,'lCs1KXiiBwi4SswcrjCkD7tH');"]

4. Versions of the cluster

    4.0.11

    - TiCDC version (execute `cdc version`):

        Release Version: v5.0.0-master
        Git Commit Hash: adcc9c6faf3d84fecd51739219304c747ee93add
        Git Branch: heads/refs/tags/v5.0.0-nightly
        UTC Build Time: 2021-03-27 16:50:21
        
[cdc.log](https://github.com/pingcap/ticdc/files/6228641/cdc.log)
Go Version: go version go1.13 linux/amd64

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Fix #1585 

### What is changed and how it works?

- Set session variable `explicit_defaults_for_timestamp` to `ON` in MySQL sink, which will keep the same behavior between upstream TiDB and downstream MySQL 5.7.
- Note only set this session variable when the value `explicit_defaults_for_timestamp`  in downstream is `OFF`, because in TiDB variable `explicit_defaults_for_timestamp` is readonly.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note

- Set session variable `explicit_defaults_for_timestamp` to `ON` to make downstream MySQL5.7 keeps the same behavior with upstream TiDB.

### CODE REVIEW

File: cdc/sink/mysql_test.go [hunk 1]
```diff
@@ -704,6 +704,29 @@ func (s MySQLSinkSuite) TestConfigureSinkURI(c *check.C) {
...
+		c.Assert(err, check.IsNil)
+		params := defaultParams.Clone()
+		dsnStr, err := configureSinkURI(context.TODO(), dsn, params, db)
+		c.Assert(err, check.IsNil)
+		expectedParams := []string{
+			"tidb_txn_mode=optimistic",
+			"readTimeout=2m",
+			"writeTimeout=2m",
+			"allow_auto_random_explicit_insert=1",
+			"explicit_defaults_for_timestamp=ON",
```
Comment (overvenus) [inline cdc/sink/mysql_test.go:717]: Could you test `on|OFF|off|1|0|❌|invalid` too?

Comment (amyangfei) [reply cdc/sink/mysql_test.go:717]: Check whether downstream is TiDB by `select version`, we don't relay the value of `explicit_defaults_for_timestamp` in database variable now. No need to verify whether these values are valid

**Review by leoppro — APPROVED** [review]

**Pull request merged**